### DISCUSSION TITLE
Há alguma restrição por parte do WebService?

### DISCUSSION THREAD

**Discussion by elvitin** [lucashpmelo/node-mde#9] [Q&A]
Boa tarde, primeiramente gostaria de agradecer muito por sua colaboração!

Eu tenho algumas dúvidas.

Há alguma restrição de requests que podem ser feitas para o WebService? que cuidados eu devo tomar ao utilizar esta API?

**Comment by lucashpmelo** [comment]
Boa noite, o Web Service de Distribuição de DF-e possui algumas restrições de consulta.

Por exemplo, consultas feitas pelo ultimo NSU devem respeitar a sequencia de NSUs retornada pela Sefaz, consultas que não seguirem a sequencia podem tomar um Consumo Indevido, nesse caso o autor fica uma hora bloqueado para novas consultas, e esse tempo é resetado toda vez que uma nova consulta é feita durante o bloqueio.

As requisições utilizando a Chave da NF-e ou o NSU também tem um limite de no máximo 20 requisições por hora.

Todas as restrições e funcionalidades estão especificadas na [Nota Técnica 2014.002](https://www.nfe.fazenda.gov.br/portal/exibirArquivo.aspx?conteudo=VgPuhtPApQo=).

**Reply by elvitin** [reply]
No caso então, a cada resposta, é me retornado um NSU, esse NSU eu tenho que salvar para utilizar em uma consulta posterior, procede?

**Reply by lucashpmelo** [reply]
Sim, isso mesmo.

**Reply by elvitin** [reply]
E quando é a primeira consulta? onde não sei o NSU, passo `000000000000000` ?

**Reply by lucashpmelo** [reply]
Exatamente, quando você não sabe qual o NSU é só informar um todo zerado, assim você obtém o próximo no retorno junto dos documentos fiscais.

**Reply by WesleyjanLacerda** [reply]
![image](https://github.com/user-attachments/assets/d9b389dd-086e-4d31-b891-58a90e77b5fe)
Olá, primeiramente muito obrigado pela a sua contribuição nesse repositório de importante utilidade.
Ali você disse que poderá ser bloqueado por uma hora caso não siga algumas dos critérios, no caso isso ocorre se eu "simplesmente" passar o último NSU errado?
Suponhamos que o último NSU é 000000000000300, e por algum "erro" foi passado "000000000000250", já é bloqueado por uma hora ?

**Reply by WesleyjanLacerda** [reply]
![image](https://github.com/user-attachments/assets/c4dc9451-ea15-4171-8978-aeefedee2342)
Acho que já achei a resposta no documento informado acima!

**Reply by lucashpmelo** [reply]
Boa tarde @WesleyjanLacerda, é exatamente isso, caso você consulte o NSU fora da ordem retornada pela Sefaz você irá receber um bloqueio de 1 hora. Por isso é importante guardar o `ultNSU` retornado em cada consulta.
